src/Model.php

```diff
@@ -1173,18 +1173,20 @@ public function reload()
      * record in the database.
      *
      * @param mixed $newId
-     *
-     * @return $this
      */
-    public function duplicate($newId = null)
+    public function duplicate($newId = null): self
     {
-        $this->setId(null);
+        $duplicate = (clone $this);
 
-        if ($this->id_field) {
-            $this->setId($newId);
+        $duplicate->dirty = $this->data;
+
+        $duplicate->entityId = null;
+
+        if ($duplicate->id_field) {
+            $duplicate->setId($newId);
```

Comment:
this is super questionable, as by "allowing ID to be set easily", we "allow to overwrite existing record with that ID by simply allowing to call save()"

Reply:
Yes, we do but this is existing functionality. We should otherwise remove the `$newId` parameter and allow saving as new record only.

Reply:
I would highly prefer that - throw if more than 0 args are given

Reply:
Done